sotodlib/coords/pmat.py

```diff
@@ -189,6 +189,9 @@ def for_tod(cls, tod, sight=None, geom=None, comps='T',
         # Set up the detectors in the focalplane
         fp = _valid_arg(focal_plane, 'focal_plane', src=tod)
         xi, eta, gamma  = fp.xi, fp.eta, fp.get('gamma')
+        if not np.all(~np.isnan([xi, eta, gamma])):
+            raise ValueError('np.nan is included in focal_plane')
+        
```

Comment:
gamma is allowed to be None here, which I think will cause the array promotion of [xi, eta, gamma] to fail.

Simpler logic uses `if np.any(np.isnan(...))`.